### ISSUE TITLE
Review #1

### ISSUE THREAD

**Issue by lukasbystricky** [Revali25/emilyofnewmoon#1]
Hi Ben,

It looks quite good overall, but there are a few things that need some fixing. I had a couple questions about your workflow though. Why did you add the chapters at different points? It would have made more sense to add them all together after splitting them. It looks like after you add them you'd already run clean, did you run anything else on them before adding them? 

In general it would be better if your commit messages were more descriptive. For example a18e30b is "Update chapter 10". It looks like you've run clean and changed some semantics, so a better commit message would be "Clean chapter 10, correct semantics". Likewise in 81d3fd3 the message "Formatted text for chapter 9", but the commit is adding chapters 8 and 9, as well as some CSS. You should also consider keeping commits to a single "type" of work. This isn't always easy to do retroactively so it's not essential, but sometimes it is straightforward. For instance b8227d4 "Updated section-id + added some chapter 10 text" should really be two separate commits (and "added some chapter 10 text" is quite vague so that should be re-worded). 

Anyways, I've made a first pass. Let me know if there's anything that's not clear or that you need help with. If you're not quite comfortable with Git it might be a bit daunting, but I don't think it's anything too extreme. Feel free to ask here or in the group chat if you have any questions. 

- [x] Run typogrify. I think most or maybe all of the changes it makes should be accepted. Do this first, the remaining review assumes this has been run.
- [x] Fix mismatched dashes. In particular there is one case of "grownup" that should be "grown-up", as it's an adjective.
- [x] Unsemanticated `<i>` around ship names. You should use se:name.vessel.ship
- [x] Poems (unless it's a book-length epic) should be quoted not italicized (see [8.2.11.3](https://standardebooks.org/manual/1.8.0/8-typography#8.2.11.3)). You can keep the semantics by wrapping it in a `span` tag. I think you should tag "Bugle Song" as a song instead of a poem. 
- [x] The sonnet "Lines to a Mountain Cascade" can also be quoted, and tagged as a poem.
- [x] "nom-de-plume" appears in Merriam Webster so you can remove the italics and language tag.
- [x] Commits 72598ce and 19fa9f5 can be dropped. This might fix the issues with added files elsewhere. 
- [ ] Commits 973e724, 85ca492: I think those quotes can be left in. (I agree with dropping them in 0ff0db5 and 0501c16) 
- [ ] Commit 1953a32: there's an extra file added here (kate.project.ctags.dc.JgrN). I guess it gets removed later because I don't see it the final repo, but you should take it out here so it's not added at all.
- [ ] Commit 7fd9618: are these all transcription errors? If so you should rename the commit, if not printer's errors need to be in their own [Editorial] commit. 
- [ ] Commit 273be1c: This should be marked as [Editorial]. You've added another file here that should be removed. If you like you can also squash this into f506b21.
- [ ] Commit a389549: It seems like this is mostly accepting corrections by the PG transcriber, which is fine, but there are other things in here. I don't think there's a mix of editorial and non-editorial so it's ok. If you're comfortable with Git you can split it, but I would at least rename it either to something more accurate like "Accept PG corrections, add semantics".

**Comment by Revali25** [comment]
Thanks for the feedback! I am new at using Git, so I was treating it more like a save function than version control, hence the more disorganized commit history. I was paranoid of losing changes so this ended up meaning that things happened out of order. It's good to learn, so I'll keep that in mind going forward.

I won't lie the step by step guide was kind of confusing for me at points, like I'm not really sure what the function of clean specifically is, besides a nebulous "it cleans up the code." I was also having issues with the python version, it wasn't stated in the step by step guide to use version 3.8-3.11, I was able to figure that part out though. Is there a recommended workflow or like, tips and tricks to efficiently find out the proper semantic inflection tags? Not every tag that I needed to use was in the manual of style, the example that comes to mind is `<i>` being untagged for ships, which I thought the manual said didn't need an inflection.

I don't know why Kate insists on putting files in the working directory instead of a separate cache location, but any file with "kate" in the name in it is from that, and I always tried to make sure it was removed before a commit; that didn't happen all the time evidently though.

I ran clean and lint before making the final commits, but not throughout the process. I'll keep that in mind going forward as well.

When I get home I'll take a look through the issues and address them.

**Comment by lukasbystricky** [comment]
Yeah there's definitely a steep learning curve. Basically what clean does is that it properly formats the html. It won't make any structural changes, so the output will look identical if you open the file in a web browser (or if you build the epub), but it makes the code easier to read. Feel free to run clean and typogrify throughout the process if you want, but make sure to run them only when you don't have any uncommitted git changes, otherwise things can get very messy. Also, if you're not already using one, I would highly recommend a graphical interface for git. I use [git fork](https://git-fork.com), but there are others. 

Semantic inflection is discussed in [Section 4.2 of the manual ](https://standardebooks.org/manual/1.8.0/single-page#4.2). The se list of element is [here](https://standardebooks.org/vocab/1.0) and you should also be aware of the [z3998 vocabulary](http://www.daisy.org/z3998/2012/vocab/structure/). Unsemanticated `<i>` tags are rather uncommon so there will usually be an appropriate inflection somewhere. 

Like I said, there's lots to learn, but I think once you get the hang of it it's not so bad. The Git stuff could be a headache especially if you're new to it, so don't hesitate to ask for any help or clarifications.

**Comment by Revali25** [comment]
I'm currently working on removing commit 72598ce, does that mean I have to revert the entire repo to before that change? if I needed to do that, would I copy the files as they are to a separate location, revert to the change, and then copy them back into the repo? I'm seeing stuff about cherry picking, would that make more sense for this?

**Comment by lukasbystricky** [comment]
So if you do [an interactive rebase](https://www.sitepoint.com/git-interactive-rebase-guide/) it should be possible to select the commit you want and drop it entirely. Looking through your commits though there's a lot of extra Kate files that have been added as parts of other commits and these will have to be edited. It can be a bit tricky, but essentially the idea will be to pick those commits, unstage the unwanted files, delete them (regular deletion, _not_ git rm), and then continue with the rebase. Make sure to make a backup copy of the repo because it's not uncommon to mess something up and then it's no longer possible to recover. 

Give that a try at least, and if you have trouble with it I can have a look.

**Comment by Revali25** [comment]
Yeah those Kate files are annoying as hell. I can't figure out what setting they're under. On my desktop at least the file name is generated randomly each time too so they have to manually be added to gitignore. I'll take a look at that tomorrow to see what I can do about them.

Ben Winstead
________________________________

**Comment by lukasbystricky** [comment]
Adding something like `kate.*` to your gitignore should work.

**Comment by lukasbystricky** [comment]
Hi Ben,

Have you made any progress on this? Let me know if you need any help with anything.

Best wishes,

Lukas

**Comment by Revali25** [comment]
Hey Lukas! Apologies for the delay, I just got back from vacation. I’ll be pushing changes later on this coming week.

Ben Winstead
________________________________

**Comment by lukasbystricky** [comment]
No need to apologize! There's no rush, just wanted to make sure there weren't any problems.

**Comment by Revali25** [comment]
I am following the guide you sent for interactive rebasing, but the command is giving me issues with that. I use the desktop app for commits and pushes and such, but for some reason when trying to run `git rebase -i` it gives me back `cannot rebase: you have unstaged changes. please commit or stash them` but when I check the app it says there's no changes made. I tried running `git push` to see if there was anything but I don't want to add something if I don't know what the changes are.

**Comment by lukasbystricky** [comment]
What does `git status` say? It might be that you've somehow created another Kate file, in which case I would update and commit the gitignore file. 

Which app do you mean? Are you using a graphical interface for git?

**Comment by Revali25** [comment]
I’m using the GitHub official desktop app. I'll take a look at that with the gitignore file.

Ben Winstead
________________________________

**Comment by Revali25** [comment]
I think I've rebased issue 7 in the list, please let me know if it's working right. With regards to commit https://github.com/Revali25/emilyofnewmoon/commit/7fd9618237b71572e0f6238a3243a622f6eea1d6, those are my errors that I saw after previously committing. I did make sure to use the [Editorial] tag liberally throughout the process, so anything that isn't marked as editorial shouldn't be editorial. 

With issue 8, do I need to do anything special? Or just leave them unchanged.

With the rest of the issues, I'll take a look here quickly. Thanks.

**Comment by lukasbystricky** [comment]
I'll look at it more closely in the next few days, but off the top of my head it doesn't quite look like I imagined it would. I know that what I wrote was much easier said than done though. If it's not right I can try and see what can be done.

**Comment by lukasbystricky** [comment]
So I created a fork of the repo and rebased it to get closer to what I was aiming for: https://github.com/lukasbystricky/emilyofnewmoon

I removed all the Kate files, split up a few of the commits and reworded some others. 

I think you can work with that repo from now on. I'd go ahead and make any remaining non-rebasing edits. There may be some additional work left on the repository side, but let's worry about that later.

**Comment by Revali25** [comment]
So that I can understand better, what were the steps you took to get it the way it is? I tried doing git cherry-pick, and git rebase -i but evidently they didn't work properly. I put the commit id as the one previous to the problematic commits.

In any case, what’s the next step on getting this done? I definitely want to just finish this haha.

Thanks,

Ben Winstead
________________________________

**Comment by lukasbystricky** [comment]
I used a graphical interface, so I don't know the exact commands, but roughly what I did was this:

1. Edit one of the early commits and add Kate.* to a .gitignore file. Amend the commit, continue rebase. This simply adds the gitignore file and should not cause any conflicts that have to be resolved. 
2. Edit all the commits involving Kate files. Unstage them (causing them to be ignored). Amend the commit, continue rebase. 
3. Edit the commits that I wanted to split. Unstage some of the files commit once, restage remaining files, amend commit. Continue rebase. 
4. Force push to remote Really make sure everything's looks as you think it should (including the history) before you do this, since you won't be able to recover the remote afterwards.

That's basically it. I think I ran into some conflicts from one of the edits, but I can't remember which. It was simple to resolve though; my editor gave me the option to choose from which commit to resolve the conflict. 

Rebasing can be a pain, but I think it's a skill worth learning if you plan on working with git at all in the future. Having a good graphical interface makes it a lot easier, especially when you're starting out. 

As for next steps, I can have another look this weekend. Hopefully there's not much left.

**Comment by lukasbystricky** [comment]
I added some issues on the new repository: [https://github.com/lukasbystricky/emilyofnewmoon/issues/1](https://github.com/lukasbystricky/emilyofnewmoon/issues/1)

If you've already cloned that repo make sure to pull because I made some additional commits/rebases. I also removed the .gitignore file because apparently lint didn't like that so be careful not to add any Kate files by accident.

**Comment by Revali25** [comment]
Ok, sounds good. When I’m making these changes, should I be pushing them to my repo? Or making changes to yours?

Ben Winstead
________________________________

**Comment by lukasbystricky** [comment]
You should push directly to the one I created. I think you have access, let
me know if not.

**Comment by Revali25** [comment]
Do you want me to make a commit for each item in the list? Or each step that I take to complete each item on the list? I don't want to over-commit again.

-Benjamin Winstead
________________________________

**Comment by lukasbystricky** [comment]
The issue before wasn't exactly that you made too many commits. If anything
it was the opposite, I had to split several of your commits.

So commit as often as you like, but the general rule is that each commit
should be the same kind of operation. Make sure that your commit message is
clear. And don't use git add -a. That should ensure you don't add any Kate
files by accident.

**Comment by Revali25** [comment]
This time I’m going to try using vs code for the edits and hope that it doesn’t add any unnecessary junk. I would have used it in the first place but I couldn’t figure out how to change just the text color 🙃

Ben Winstead
________________________________
